Source/engine/palette.cpp

```diff
@@ -291,7 +291,7 @@ void PaletteFadeIn(int fr, const std::array<SDL_Color, 256> &srcPalette)
 			RenderPresent();
 		}
 	}
-	UpdateSystemPalette(palette);
+	UpdateSystemPalette(srcPalette);

```

Comment:
Alternatively, if we don't mind bypassing the `UpdateSystemPalette()` abstraction for a negligible bit of speed...
```suggestion
	memcpy(system_palette.data(), palette, system_palette.size());
	SystemPaletteUpdated();
	RedrawEverything();
```

Reply:
Yeah I think bypassing the abstraction would be good here, for clarity more so than performance.

Reply:
Done